File: FrontEnd/GitHubSimulatorUI/src/app/modules/repositories/model/Repository.ts
import { Visibility } from "./Visibility";

export class Repository {
  id: string = '';
  name: string = '';
  description: string = '';
  visibility: Visibility = Visibility.Private;
  stars_Count: number = 0;
  forks_Count: number = 0;
  watchers_Count: number = 0;
  created_At: Date = new Date();
  updated_At: Date = new Date();
  clone_Url: string = '';
  ssh_Url: string = '';
}

File: BackEnd/GitHubSimulator.Infrastructure/Models/Repository.cs
using GitHubSimulator.Core.Models.Enums;

namespace GitHubSimulator.Infrastructure.Models;

public class Repository
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public Visibility Visibility { get; set; }
}
